### ISSUE TITLE
Nightly issue: dioxus-cli fails to build due to new Cargo.lock (version 4) format

### ISSUE THREAD

**Issue by alex-tianhuang** [DioxusLabs/dioxus#3034]
**Problem**

When starting a new dioxus project using dioxus-cli compiled on nightly, the build fails as the Cargo.lock version (=4) is newer than expected. Additionally, using dioxus-cli compiled from stable with any project that has this new Cargo.lock version (I'm trying to use it for a project which requires nightly)

**Steps To Reproduce**

1. Install dioxus v0.5.7  using cargo install (nightly-aarch64-apple-darwin rustc)
2. Make a project using dx new + dx serve
OR
1. Install dioxus v0.5.7  using cargo install (stable rustc)
2. Run dx serve on an existing project which uses Cargo.lock version 4

**Expected behavior**

Dioxus serves the webpage

**Screenshots**
Here are all the steps needed to reproduce the bug:
<img width="999" alt="image" src="https://github.com/user-attachments/assets/336b6082-494c-49d4-b228-22f4d4b4c207">
Here is the final failing build command with some traceback:
<img width="998" alt="image" src="https://github.com/user-attachments/assets/3274d6ea-bac1-4b06-92de-5802e1622484">
Here is a Cargo.lock generated by dx new (nightly compiled dioxus-cli):
<img width="643" alt="image" src="https://github.com/user-attachments/assets/5dbf53a2-42fa-4821-8a1a-4abe72c340a8">

**Environment:**
 - Dioxus version:  0.5.7
 - Rust version:    1.83.0-nightly
 - OS info:  MacOS

**Comment by OlePaycxEu** [comment]
Same problem here.

**Environment:**

- Dioxus version: 0.5.6
- Rust version: 1.83.0-nightly
- OS info: Windows 10

**Comment by GTime** [comment]
Same problem here. 

**Environment**:
- Dioxus 0.5

```sh
dx serve --platform fullstack
  👑 Build done.
| 💼 Waiting to start building the project...                                                 thread 'main' panicked at /home/siki/.cargo/registry/src/index.crates.io-6f17d22bba15001f/manganis-cli-support-0.2.4/src/manifest.rs:51:51:
called `Result::unwrap()` on an `Err` value: Parse("parse error: invalid Cargo.lock format version: `4`\n")
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by msukanen** [comment]
Same boat here...

Dioxus 0.5.6
Rust 1.84.0-nightly
OS: Win11

**Comment by luis-prates** [comment]
I'm having the same issue. I got the build working by manually changing the Cargo.lock version to 3.

- rustc: 1.83.0-nightly (1bc403daa 2024-10-11)
- dioxus 0.5.7
- Win 11

Running BeforeDevCommand (`dx serve --port 1420`)
    Warn Waiting for your frontend dev server to start on http://localhost:1420/...
  👑 Build done.                                                                                                                                                                           
thread 'main' panicked at C:\Users\mine\.cargo\registry\src\index.crates.io-6f17d22bba15001f\manganis-cli-support-0.2.4\src\manifest.rs:51:51:
called `Result::unwrap()` on an `Err` value: Parse("parse error: invalid Cargo.lock format version: `4`\n")
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
    Error The "beforeDevCommand" terminated with a non-zero status code.

**Comment by DogeDark** [comment]
I did a little digging and this issue might be fixed in the 0.6 alpha CLI. If you can't use the 0.6 alpha version of Dioxus due to bugs or other reasons, you can see if what @luis-prates said works, and manually set the lock version to `3`.

`cargo install dioxus-cli@0.6.0-alpha.3`
and
`dioxus = { version = "0.6.0-alpha.3", features = ["myplatformhere"] }`

**Comment by zanekwok** [comment]
some

thread 'main' panicked at /Users/peguo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/manganis-cli-support-0.2.4/src/manifest.rs:51:51:
called `Result::unwrap()` on an `Err` value: Parse("parse error: invalid Cargo.lock format version: `4`\n")
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

**Comment by dxps** [comment]
Today, I updated Rust to the new release ver. 0.83 and now I'm facing this issue for the first time on [this sample](https://github.com/dxps/dioxus_playground/tree/main/dioxus_05_list-dnd):

```shell
❯ ./run_dev.sh 
  👑 Build done.                                                                                                                                                                   thread 'main' panicked at /home/dxps/.cargo/registry/src/index.crates.io-6f17d22bba15001f/manganis-cli-support-0.2.4/src/manifest.rs:51:51:
called `Result::unwrap()` on an `Err` value: Parse("parse error: invalid Cargo.lock format version: `4`\n")
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: <manganis_common::manifest::AssetManifest as manganis_cli_support::manifest::AssetManifestExt>::load_from_path
   4: dioxus_cli::assets::asset_manifest
   5: dioxus_cli::builder::build_web
   6: dioxus_cli::server::web::serve::{{closure}}
   7: dioxus_cli::cli::serve::Serve::serve::{{closure}}
   8: dx::main::{{closure}}
   9: tokio::runtime::park::CachedParkThread::block_on
  10: tokio::runtime::runtime::Runtime::block_on
  11: dx::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
❯ 
```

Removed `Cargo.lock`, and did all the cleanup (`dx clean`, `cargo clean`), same result.

**Comment by ealmloff** [comment]
This is fixed in the 0.6.0-alpha. We no longer read the `Cargo.lock` file at all. Instead we look for assets in the link section of the binary cargo produces

**Comment by davber** [comment]
> This is fixed in the 0.6.0-alpha. We no longer read the `Cargo.lock` file at all. Instead we look for assets in the link section of the binary cargo produces

Doesn't seem fixed in that version, nor the RC0 version. And, using the stable channel for Rust(up) doesn't suffice. One has to downgrade to 1.82 to get back to lock version 3.

**Comment by wiseaidev** [comment]
I recently came across the issue mentioned by @GTime . The reason for this is that the latest Rust toolchain, 1.85, is incompatible with Dioxus 0.5.6.

One way to resolve this is by pinning the Rust toolchain to version 1.80 and downgrading the crates used in your project.

In case it helps, I've fixed similar issues in https://github.com/opensass/opensass/pull/23.
